### ISSUE TITLE
Crash in BraveSyncServiceImpl::GetOrCreateSyncCode

### ISSUE THREAD

**Issue by iefremov** [brave/brave-browser#20303]
https://brave.sp.backtrace.io/p/brave/debug?time=month&filters=((callstack.functions%2Ccontains%2CGetOrCreateSyncCode)%2C_deleted%3D0%2Cptype%3Dbrowser)&debug=(%222dabfb7%22,0,0)
```
[ 00 ] syncer::BraveSyncServiceImpl::GetOrCreateSyncCode()
[ 01 ] BraveSyncHandler::HandleGetSyncCode(base::CheckedContiguousRange<std::__1::vector<base::Value, std::__1::allocator<base::Value> > const>)
[ 02 ] content::WebUIImpl::ProcessWebUIMessage(GURL const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, base::ListValue const&)
[ 03 ] content::WebUIImpl::Send(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, base::Value)
[ 04 ] content::mojom::WebUIHostStubDispatch::Accept(content::mojom::WebUIHost*, mojo::Message*)

```

**Comment by iefremov** [comment]
cc @AlexeyBarabash @jsecretan

**Comment by darkdh** [comment]
`OSCrypt::DecryptString` in `brave_sync::crypto::GetSeed()` would fail easily if users don't give it permission. ex. an user rejected the request to access the "Brave Safe Storage" in keychain.

**Comment by BugRMSys** [comment]
> Happened on `CHECK` at
> 
> ```
> std::string BraveSyncServiceImpl::GetOrCreateSyncCode() {
>   std::string sync_code = brave_sync_prefs_.GetSeed();
>   if (sync_code.empty()) {
>     std::vector<uint8_t> seed = brave_sync::crypto::GetSeed();
>     sync_code = brave_sync::crypto::PassphraseFromBytes32(seed);
>   }
> 
>   CHECK(brave_sync::crypto::IsPassphraseValid(sync_code))
>       << "Could not generate valid sync code";
> 
>   return sync_code;
> }
> ```
> 
> This `CHECK` should not be reached. Something went wrong. I am sure for `brave_sync::crypto::GetSeed()` and `brave_sync::crypto::PassphraseFromBytes32` , so maybe somehow the wrong value was written into the prefs and `brave_sync_prefs_.GetSeed()` gave non-empty value.

Please fix the bug #20739

**Comment by stephendonner** [comment]
Verification **PASSED** on `macOS` using the following build:


Went through the STR/Cases outlined via https://github.com/brave/brave-core/pull/11935#issue-1109703137 and ensured the following:

Created a new sync chain between `Desktop` & `Android` and ensured that `brave://sync-internals` looked good when initially joining and when enabling `Sync All`:

**`Default Settings (initially joining)`** | **`Enabling "Sync All"`**
------------------------------------------|-----------------------------
<img width="336" alt="Screen Shot 2022-02-11 at 11 54 34 AM" src="https://user-images.githubusercontent.com/387249/153680231-f71647a4-8c12-41d2-9ff3-234dd6b73cde.png"> | 
<img width="330" alt="Screen Shot 2022-02-11 at 11 56 28 AM" src="https://user-images.githubusercontent.com/387249/153680245-267aacb3-9fa9-4df1-bc23-2b3343abbe08.png">




Once the above was confirmed, left the sync chain and re-joined and ensured that everything appeared green after enabling `Sync All` again:

<img width="330" alt="Screen Shot 2022-02-11 at 11 56 28 AM" src="https://user-images.githubusercontent.com/387249/153680257-70571370-5ee3-4b58-b2e0-bf76891e15f5.png">


Removed/Disabled sync on the desktop machine via `Disable Sync (Clear Data)` and ensured that everything was disconnected as per the following:

**`Example`** | **`Example`**
---------------|-----------------
<img width="1625" alt="Screen Shot 2022-02-11 at 11 57 24 AM" src="https://user-images.githubusercontent.com/387249/153680288-3f07a403-b0ee-4ef9-8f7a-9a61b62eedd7.png"> | <img width="1069" alt="Screen Shot 2022-02-11 at 11 57 35 AM" src="https://user-images.githubusercontent.com/387249/153680292-ba27f75f-fa70-45d4-8bea-89fc0d4ef8fc.png">
